### ISSUE TITLE
Single Path One-Shot Neural Architecture Search with Uniform Sampling

### ISSUE THREAD

**Issue by sile** [sile/readings#106]
URL
-----

https://arxiv.org/abs/1904.00420

Abstract
----------

> One-shot method is a powerful Neural Architecture Search (NAS) framework, but its training is non-trivial and it is difficult to achieve competitive results on large scale datasets like ImageNet. In this work, we propose a Single Path One-Shot model to address its main challenge in the training. Our central idea is to construct a simplified supernet, Single Path Supernet, which is trained by an uniform path sampling method. All underlying architectures (and their weights) get trained fully and equally. Once we have a trained supernet, we apply an evolutionary algorithm to efficiently search the best-performing architectures without any fine tuning. Comprehensive experiments verify that our approach is flexible and effective. It is easy to train and fast to search. It effortlessly supports complex search spaces (e.g., building blocks, channel, mixed-precision quantization) and different search constraints (e.g., FLOPs, latency). It is thus convenient to use for various needs. It achieves start-of-the-art performance on the large dataset ImageNet.

Table of Contents
--------------------

- (1) Introduction
- (2) Review of NAS Approaches
  - Weight Sharing Approaches
- (3) Our One-Shot NAS Approach
  - (3-1) Revisiting One-Shot NAS
  - (3-2) Single Path Supernet and Uniform Sampling
  - (3-3) Supernet Architecture and Choice Blocks
    - Channel Number Search
    - Mixed-Precision Quantization Search
  - (3-4) Evolutionary Architecture Search
  - (3-5) Summary
- (4) Experiment Results
  - Dataset
  - Training
  - Search Space: Building Blocks
  - Search Space: Channels
  - Comparison with State-of-the-arts
  - Application: Mixed-Precision Quantization
  - Search Cost Analysis

BibTeX
------

```bibtex
@misc{guo2019single,
    title={Single Path One-Shot Neural Architecture Search with Uniform Sampling},
    author={Zichao Guo and Xiangyu Zhang and Haoyuan Mu and Wen Heng and Zechun Liu and Yichen Wei and Jian Sun},
    year={2019},
    eprint={1904.00420},
    archivePrefix={arXiv},
    primaryClass={cs.CV}
}
```

**Comment by sile** [comment]
>  the statistics of all the Batch Normalization (BN) [11] operations are recalculated on a random subset of training data (20000 images on ImageNet). It takes a few seconds. This is because the BN statistics from the supernet are usually not applicable to the candidate nets.
